environment/src/main/java/com/sequenceiq/environment/environment/service/EnvironmentCreationService.java

```diff
@@ -123,6 +126,15 @@ private Environment initializeEnvironment(EnvironmentCreationDto creationDto) {
         return environment;
     }
 
+    private void initializeEnvironmentTunnel(Environment environment) {
+        Tunnel tunnel = environment.getExperimentalFeaturesJson().getTunnel();
+        if (Tunnel.CCM == tunnel && entitlementService.ccmV2Enabled(environment.getAccountId())) {
+            ExperimentalFeatures experimentalFeaturesJson = environment.getExperimentalFeaturesJson();
+            experimentalFeaturesJson.setTunnel(Tunnel.CCMV2);
+            environment.setExperimentalFeaturesJson(experimentalFeaturesJson);
+        }
+    }
```

Comment:
So you shouldn't overwrite CCM with CCMv2. You should check the request, if it contains CCMv2 but the entitlement is disabled you must throw a badrequestexception with a reason.
If CCM is sent in the request,it should be used, even if the entitlement is enabled. We don't want to force users with enabled entitlement to use CCMv2, do we?

Reply:
yes I think that is expectation when entitlement is enabled only CCMv2 is to be used. Also CLI only passes _**enableTunnel**_ option there is no CCM or DIRECT option exposed to users.

Reply:
ok, I get it, but still, you should check that you accept CCMv2 only when the entitlement enabled. We can't rely on the logic present in other component. I think the UI eg communicates directly and sends CCM.

Reply:
But that would mean we will not able to create CCMv2 clusters via CDP CLI because it just as "enableTunnel", I think that would be a bigger issue.

Reply:
I mean you can leave this part, if CCM is received and entitlement enabled, turn it into CCMv2. so this part of the code is fine.
But you should add a validation before this, if CCMv2 is received and the entitlement is disabled then return a badrequest.

Reply:
Added validation if tunnel is CCMV2 and not entitled, badrequestexception is thrown.